Source/WebCore/platform/graphics/PlatformTimeRanges.cpp

```diff
@@ -132,6 +132,35 @@ MediaTime PlatformTimeRanges::maximumBufferedTime() const
     return m_ranges[length() - 1].m_end;
 }
 
+unsigned int PlatformTimeRanges::getNearestSmallerStart(const MediaTime& start, const MediaTime& end) const
+{
+    ASSERT(start <= end);
+    Range range(start, end);
+    unsigned first, last, middle;
```

Comment:
All the indexes should be size_t. PlatformTimeRanges::length() is unsigned, but it's a bit weird and I wouldn't use it as a reason to choose unsigned int as the type. Better do like find() and use size_t, the actual index type of Vector. maybe you need to declare "unsigned overlappingArcIndex;" in ::add() also as size_t too.